src/main/java/Duke.java

```diff
@@ -1,10 +1,331 @@
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.File;
+import java.io.IOException;
+import task.*;
+
+class DukeException extends Exception {
+    public DukeException(String errorMessage) {
+        super(errorMessage);
+    }
+}
+
 public class Duke {
-    public static void main(String[] args) {
+
+    public static ArrayList<Task> allTasks = new ArrayList<>();
+
+    private static void appendToFile(String filePath, String textToAppend) throws IOException {
+        FileWriter fw = new FileWriter(filePath, true); // create a FileWriter in append mode
+        fw.write(textToAppend);
+        fw.close();
+    }
+
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    private static void ReadFile(File fileObj) throws FileNotFoundException{
+        try {
+            Scanner fileReader = new Scanner(fileObj);
+            while (fileReader.hasNextLine()) {
+                String line = fileReader.nextLine();
+                char taskType = line.charAt(4);
+                char doneStatus = line.charAt(7);
+                String subLine = line.substring(10);
+                switch (taskType) {
+                    case 'T':
+                        Todo t = new Todo(subLine);
```

Comment:
Consider giving a more descriptive name to your variables. Similar issue in other places as well.